src/main/java/games/strategy/util/PredicateBuilder.java

```diff
@@ -0,0 +1,57 @@
+package games.strategy.util;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.function.Predicate;;
+
+public class PredicateBuilder<T> {
+
+  private Predicate<T> predicate;
+
+  private PredicateBuilder(final Predicate<T> predicate) {
+    this.predicate = checkNotNull(predicate);
+  }
+
+  /**
+   * Creates a new PredicateBuilder starting with the provided Predicate.
+   */
+  public static <T> PredicateBuilder<T> of(final Predicate<T> predicate) {
+    return new PredicateBuilder<>(predicate);
+  }
+
+  /**
+   * Modifies this PredicateBuilder to return the underlying predicate
+   * combined with a logical AND with the given Predicate.
+   */
+  public PredicateBuilder<T> and(final Predicate<T> predicate) {
```

Comment:
This may be a YAGNI, but I'm not sure if the parameters to `and()`, `andIf()`, `or()`, and `orIf()` should be `Predicate<? super T>` to match the capabilities of `Predicate#and()` and `Predicate#or()`.

Nit: Also, renaming the parameters to `other`, as the `Predicate` methods do, would avoid having to qualify `this.predicate` where it is currently necessary.